Cron/Processor.php

```diff
@@ -186,8 +195,12 @@ private function updateItems($store, $itemType, $indice)
     private function deleteItems($store, $itemType, $indice)
     {
         $collection = $this->changedItemCollectionFactory->create()->filterDeleted((int)$store->getId(), $itemType);
-        if ($collection->getSize()) {
-            $deleted = $this->documentsProvider->getBatched($collection);
+
+        $groupedCollection = clone $collection;
+        $groupedCollection->getSelect()->group(ChangedItemInterface::ITEM_ID);
+
+        if ($groupedCollection->getSize()) {
+            $deleted = $this->documentsProvider->getBatched($groupedCollection);
```

Comment:
I don't understand why this code is necessary :thinking: could you clarify it for me please?

Reply:
@yuloma entiendo que esto es algo del fixture no? no tiene mucho sentido que el elemento se cree una vez pero se nos inserte muchas, no sabemos esto por qué pasa realmente no? solo curiosidad

Reply:
y por qué de normal no falla? porque había un `exists` en algun sitio y si ya existía no se metía no? entiendo que ese exist por algún motivo no se está aplicando no?

Reply:
No he podido probar la importación de archivos csv porque no he encontrado alguno apropiado para probar. Pero si he probado generando datos con la vía que usamos habitualmente 
```
bin/magento sampledata:deploy
bin/magento setup:upgrade
bin/magento setup:di:compile
bin/magento setup:static-content:deploy -f
```
Luego de hacer la prueba veo que se capturan los eventos que sobre los productos, y se agregan a nuestra tabla. Con lo cual se agregarán elementos duplicados como en el caso anterior con los fixtures. Entiendo que pasará lo mismo con una importación normal usando un CSV desde el panel de admin.

Reply:
gracias por la confirmación @yuloma, entonces ese problema lo estábamos teniendo siempre y no teníamos conocimiento...
No se puede hacer entonces de ninguna forma para ignorar esos antes de incluirlo en el lote a insertar en base de datos porque es algo más interno de magento, no?

Reply:
Capaz y si se pueda hacer algo, pero necesitaría investigarlo, Magento usa Doctrine y seguro tiene alguna manera de manejar este tipo de casos. Pero si es algo interno de ellos, las querys nuestras se van metiendo en la trasacción también.